pkgs/development/tools/omnisharp-roslyn/default.nix

```diff
@@ -62,7 +62,7 @@ let finalPackage = buildDotnetModule rec {
           send_error "timeout!\n"
           exit 1
         }
-        expect ".NET Core SDK ${if sdk ? version then sdk.version else sdk_6_0.version}"
+        expect ".NET Core SDK ${if sdk ? version then sdk.version else sdk_8_0.version}"
```

Comment:
Afterthought but `sdk.version or sdk_8_0.version` should be equivalent

Reply:
This is called 4 times to test against 6/7/8/no-sdk and `sdk` here is the one we're testing.

Reply:
Sorry, I'm not sure I understand. But syntax-wise I think the two expressions should amount to the same thing

Reply:
Maybe I misunderstood you.  What do you propose changing it to exactly?

Reply:
`if sdk ? version then sdk.version else sdk_8_0.version` = `sdk.version or sdk_8_0.version` = the sdk's attribute "version" if it exists, otherwise sdk_8_0's attribute version

Reply:
Oh, I'm sorry, I read your first message on my phone where the code block highlighting didn't show up.  I thought you were saying `sdk.version` and `sdk_8_0.version` were equivalent.  I see what you mean now.

Yeah, we should be able to do this:

```
diff --git a/pkgs/development/tools/omnisharp-roslyn/default.nix b/pkgs/development/tools/omnisharp-roslyn/default.nix
index db7ad7d0b284..c25848bc5de1 100644
--- a/pkgs/development/tools/omnisharp-roslyn/default.nix
+++ b/pkgs/development/tools/omnisharp-roslyn/default.nix
@@ -53,7 +53,7 @@ let finalPackage = buildDotnetModule rec {
   executables = [ "OmniSharp" ];
 
   passthru.tests = let
-    with-sdk = sdk: runCommand "with-${if sdk ? version then sdk.version else "no"}-sdk"
+    with-sdk = sdk: runCommand "with-${sdk.version or "no"}-sdk"
       { nativeBuildInputs = [ finalPackage sdk expect ]; meta.timeout = 60; } ''
       HOME=$TMPDIR
       expect <<"EOF"
@@ -62,7 +62,7 @@ let finalPackage = buildDotnetModule rec {
           send_error "timeout!\n"
           exit 1
         }
-        expect ".NET Core SDK ${if sdk ? version then sdk.version else sdk_8_0.version}"
+        expect ".NET Core SDK ${sdk.version or sdk_8_0.version}"
         expect "{\"Event\":\"started\","
         send \x03
         expect eof
```

I just verified that the derivations are the same.